contrib/dropwizard/src/main/java/io/opencensus/contrib/dropwizard/DropWizardMetrics.java

```diff
@@ -0,0 +1,337 @@
+  static final List<LabelValue> RATE_MEAN_LABEL_VALUE =
+      Collections.unmodifiableList(Collections.singletonList(LabelValue.create("mean_rate")));
+
+  @DefaultVisibilityForTesting
+  static final List<LabelValue> RATE_ONE_MINUTE_LABEL_VALUE =
+      Collections.unmodifiableList(Collections.singletonList(LabelValue.create("m1_rate")));
+
+  @DefaultVisibilityForTesting
+  static final List<LabelValue> RATE_FIVE_MINUTE_LABEL_VALUE =
+      Collections.unmodifiableList(Collections.singletonList(LabelValue.create("m5_rate")));
+
+  @DefaultVisibilityForTesting
+  static final List<LabelValue> RATE_FIFTEEN_MINUTE_LABEL_VALUE =
+      Collections.unmodifiableList(Collections.singletonList(LabelValue.create("m15_rate")));
+
+  private final List<com.codahale.metrics.MetricRegistry> metricRegistryList;
+  private final Clock clock;
+  private final Timestamp cumulativeStartTimestamp;
+
+  /**
+   * Hook the Dropwizard registry into the OpenCensus registry.
+   *
+   * @param metricRegistryList a list of {@link com.codahale.metrics.MetricRegistry}s.
+   * @since 0.17
+   */
+  public DropWizardMetrics(List<com.codahale.metrics.MetricRegistry> metricRegistryList) {
+    Utils.checkNotNull(metricRegistryList, "metricRegistryList");
+    Utils.checkListElementNotNull(metricRegistryList, "metricRegistryList");
+    this.metricRegistryList = metricRegistryList;
+    clock = MillisClock.getInstance();
+    cumulativeStartTimestamp = clock.now();
+  }
+
+  /**
+   * Returns a {@code Metric} collected from {@link Gauge}.
+   *
+   * @param dropwizardName the metric name.
+   * @param gauge the gauge object to collect.
+   * @return a {@code Metric}.
+   */
+  @SuppressWarnings("rawtypes")
+  private @Nullable Metric collectGauge(String dropwizardName, Gauge gauge) {
+    String metricName = DropWizardUtils.generateFullMetricName(dropwizardName, "gauge");
+    String metricDescription = DropWizardUtils.generateFullMetricDescription(dropwizardName, gauge);
+
+    // Figure out which gauge instance and call the right method to get value
+    Type type;
+    Value value;
+
+    Object obj = gauge.getValue();
+    if (obj instanceof Number) {
+      type = Type.GAUGE_DOUBLE;
+      value = Value.doubleValue(((Number) obj).doubleValue());
+    } else if (obj instanceof Boolean) {
+      type = Type.GAUGE_INT64;
+      value = Value.longValue(((Boolean) obj) ? 1 : 0);
+    } else {
+      // Ignoring Gauge (gauge.getKey()) with unhandled type.
+      return null;
+    }
+
+    MetricDescriptor metricDescriptor =
+        MetricDescriptor.create(
+            metricName, metricDescription, DEFAULT_UNIT, type, Collections.<LabelKey>emptyList());
+    TimeSeries timeSeries =
+        TimeSeries.create(
+            Collections.<LabelValue>emptyList(),
+            Collections.singletonList(Point.create(value, clock.now())),
+            null);
+    return Metric.create(
+        metricDescriptor, Collections.unmodifiableList(Collections.singletonList(timeSeries)));
```

Comment:
The singletonList is already immutable so no need to wrap it using unmodifiableList see here:
https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#singletonList(T)

Same comment everywhere.

Reply:
Make Sense. Done